Create a better exception message when Quartz is used

**Description**

With a pure programmatic Quartz approach, with a code like:

```
@ApplicationScoped
public class PeriodicTask {

@Inject
Scheduler quartz;

public void start() throws SchedulerException {
final JobDetail job = newJob(MyJob.class)
.withIdentity("questions", "millionaire")
.build();

final SimpleTrigger trigger = newTrigger()
.withIdentity("questionsTrigger", "millionaire")
.startNow()
.withSchedule(
SimpleScheduleBuilder.simpleSchedule()
.withIntervalInSeconds(20)
.withRepeatCount(5))
.build();

quartz.getListenerManager()
.addTriggerListener(new EndTriggerListener("EndTrigger"),
keyEquals(triggerKey("questionsTrigger", "millionaire")));
quartz.scheduleJob(job, trigger);
}
}
```
You get the next exception:

```
Caused by: java.lang.IllegalStateException: Cannot produce org.quartz.Scheduler - Quartz scheduler is disabled or no schedules were found
	at io.quarkus.quartz.runtime.QuartzScheduler.produceQuartzScheduler(QuartzScheduler.java:69)
```

When you add the next property:

```
quarkus.quartz.force-start=true
```

Then everything works as expected, so I think that it would be nice to have a better error message saying, I couldn't generate the Scheduler, maybe you have not set the property to true? So people can figure out quickly what is the problem.

--- a/docs/src/main/asciidoc/scheduler-reference.adoc
+++ b/docs/src/main/asciidoc/scheduler-reference.adoc
@@ -207,6 +207,8 @@ class MyJobs {
 }
 ----
 
+NOTE: By default, the scheduler is not started unless a `@Scheduled` business method is found. You may need to force the start of the scheduler for "pure" programmatic scheduling. See also <<quartz.adoc#quartz-configuration-reference,Quartz Configuration Reference>>.
+
 == Scheduled Methods and Testing
 
 It is often desirable to disable the scheduler when running the tests.

--- a/extensions/quartz/deployment/src/test/java/io/quarkus/quartz/test/DisabledSchedulerTest.java
+++ b/extensions/quartz/deployment/src/test/java/io/quarkus/quartz/test/DisabledSchedulerTest.java
@@ -1,7 +1,10 @@
 package io.quarkus.quartz.test;
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
 
+import javax.enterprise.inject.Instance;
 import javax.inject.Inject;
 
 import org.jboss.shrinkwrap.api.ShrinkWrap;
@@ -17,7 +20,10 @@ import io.quarkus.test.QuarkusUnitTest;
 public class DisabledSchedulerTest {
 
     @Inject
-    Scheduler quartzScheduler;
+    Scheduler scheduler;
+
+    @Inject
+    Instance<org.quartz.Scheduler> quartzScheduler;
 
     @RegisterExtension
     static final QuarkusUnitTest test = new QuarkusUnitTest()
@@ -28,7 +34,13 @@ public class DisabledSchedulerTest {
 
     @Test
     public void testNoSchedulerInvocations() throws InterruptedException {
-        assertFalse(quartzScheduler.isRunning());
+        assertFalse(scheduler.isRunning());
+        assertTrue(quartzScheduler.isResolvable());
+        try {
+            quartzScheduler.get();
+            fail();
+        } catch (IllegalStateException expected) {
+        }
     }
 
     static class Jobs {

--- a/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java
+++ b/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java
@@ -71,7 +71,7 @@ public class QuartzScheduler implements Scheduler {
     org.quartz.Scheduler produceQuartzScheduler() {
         if (scheduler == null) {
             throw new IllegalStateException(
-                    "Cannot produce org.quartz.Scheduler - Quartz scheduler is disabled or no schedules were found");
+                    "Quartz scheduler is either explicitly disabled through quarkus.scheduler.enabled=false or no @Scheduled methods were found. If you only need to schedule a job programmatically you can force the start of the scheduler via quarkus.quartz.force-start=true");
         }
         return scheduler;
     }

